<launch>
    <arg name="ismovebase"  default="false"/>

    <node name="slam_gmapping" pkg="gmapping" type="slam_gmapping" output="screen">
        <remap from="scan"                     to="sensor_laser/scan"/>
        <param name="base_frame"               value="base_footprint"/>
        <param name="odom_frame"               value="odom"/>
        <param name="map_frame"                value="map"/>
        <param name="transform_publish_period" value="-1"/>
        <param name="map_update_interval"      value="2"/>
        <param name="maxUrange"                value="10.0"/>
        <param name="sigma"                    value="0.05"/>
        <param name="kernelSize"               value="1"/>
        <param name="lstep"                    value="0.05"/>
        <param name="astep"                    value="0.05"/>
        <param name="iterations"               value="5"/>
        <param name="lsigma"                   value="0.075"/>
        <param name="ogain"                    value="3.0"/>
        <param name="lskip"                    value="0"/>
        <param name="srr"                      value="0.1"/>
        <param name="srt"                      value="0.2"/>
        <param name="str"                      value="0.1"/>
        <param name="stt"                      value="0.2"/>
        <param name="minimumScore"             value="0.0"/>
        <param name="linearUpdate"             value="0.1"/>
        <param name="angularUpdate"            value="0.01"/>
        <param name="temporalUpdate"           value="0.0"/>
        <param name="resampleThreshold"        value="0.1"/>
        <param name="particles"                value="1"/>
        <param name="xmin"                     value="-15.0"/>
        <param name="ymin"                     value="-15.0"/>
        <param name="xmax"                     value="15.0"/>
        <param name="ymax"                     value="15.0"/>
        <param name="delta"                    value="0.1"/>
        <param name="llsamplerange"            value="0.01"/>
        <param name="llsamplestep"             value="0.01"/>
        <param name="lasamplerange"            value="0.005"/>
        <param name="lasamplestep"             value="0.005"/>
    </node>

    <node if="$(arg ismovebase)" pkg="move_base" type="move_base" respawn="false" name="move_base">
        <rosparam file="$(find rtcrobot)/config/costmap_common_params.yaml"     command="load" ns="global_costmap" /> 
        <rosparam file="$(find rtcrobot)/config/costmap_common_params.yaml"     command="load" ns="local_costmap" />
        <rosparam file="$(find rtcrobot)/config/local_costmap_params.yaml"      command="load" />
        <rosparam file="$(find rtcrobot)/config/global_costmap_params.yaml"     command="load" /> 
        <rosparam file="$(find rtcrobot)/config/base_local_planner_params.yaml" command="load" />
        <rosparam file="$(find rtcrobot)/config/move_base_params.yaml"          command="load" />

        <!--<param name="base_global_planner" type="string" value="navfn/NavfnROS" /> -->
        <param name="base_local_planner"                value="base_local_planner/TrajectoryPlannerROS"/>  
        </node>
</launch>
